### PR TITLE
revert InstanceMetadataByProviderID definition and related deprecations

### PR DESCRIPTION
**Pull request by gongguan**
Ref [#90652](https://github.com/kubernetes/kubernetes/issues/90652)

/kind cleanup

**Special notes for your reviewer**:
See https://github.com/kubernetes/kubernetes/pull/90894#issuecomment-633707002

```release-note
None
```

### CODE REVIEW

File: pkg/volume/cinder/attacher_test.go [hunk 1]
```diff
@@ -734,10 +734,6 @@ func (instances *instances) InstanceShutdownByProviderID(ctx context.Context, pr
 	return false, errors.New("unimplemented")
 }
 
-func (instances *instances) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
```
Comment (wojtek-t) [inline pkg/volume/cinder/attacher_test.go:737]: I wouldn't personally remove the implementations - they will follow the v2 interface as soon as this is introduced then.

I would just remove the changes to cloud.go in the interface.

Comment (gongguan) [reply pkg/volume/cinder/attacher_test.go]: Done, PTAL.

Comment (andrewsykim) [reply pkg/volume/cinder/attacher_test.go]: yeah agreed with @wojtek-t on this

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

Thanks for reverting

**Pull request merged**